values/gitea/gitea-raw.gotmpl

```diff
@@ -37,3 +39,104 @@ resources:
     S3_STORAGE_ACCOUNT: "{{ $obj.linode.accessKeyId | b64enc }}"
     S3_STORAGE_KEY: "{{ $obj.linode.secretAccessKey | b64enc }}"
 {{- end }}
+# Application backup resources
+- apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    name: gitea-backup
+  spec:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        {{- if eq $v.cluster.provider "vultr" }}
+        storage: 10Gi
+        {{- else }}
+        storage: 1Gi
+        {{- end }}
+{{- if $giteaBackupConfig.enabled }}
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    name: gitea-backup
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: gitea-backup-operator
+  rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "watch", "list"]  
+  - apiGroups: [""]
+    resources: ["pods/exec"]
+    verbs: ["create"]
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: gitea-backup
+  subjects:
+  - kind: ServiceAccount
+    name: gitea-backup
+  roleRef:
+    kind: Role 
+    name: gitea-backup-operator
+    apiGroup: rbac.authorization.k8s.io
+- apiVersion: batch/v1
+  kind: CronJob
+  metadata:
+    name: gitea-backup-job
+  spec:
+    schedule: {{ $giteaBackupConfig.schedule | quote }}
+    concurrencyPolicy: Forbid
+    jobTemplate:
+      spec:
+        template:
+          metadata:
+            annotations:
+              sidecar.istio.io/inject: "false"
+          spec:
+            serviceAccountName: gitea-backup
+            containers:
+              - image: bitnami/kubectl:1.30
+                name: kubectl
+                command:
+                  - /bin/sh
+                  - -ec
+                  - >-
+                    kubectl exec gitea-0 -- /bin/sh -ec "
+                    if [ ! -f '/backup/.bin/rclone' ]; then
+                      echo 'Installing RClone...' &&
```

Comment:
We can maybe build our own image bitnami/kubectl:1.30 and add rclone to it. This way you don't need to install it every time the cronjob is ran.

Reply:
No, this will not solve it. The Gitea container needs access to the backup and is therefore also running Rclone, not the Cronjob/Kubectl one. Maintaining our own Gitea image with Rclone is probably not worth the effort. A more "clean" but way more complex solution would be to add a sidecar to the Gitea container. However, that would require modifying the upstream chart, which I have been trying to avoid.

Note also, that Rclone is only installed on the first cronjob run.

Reply:
Oh I totally read over the kubectl exec. Was already wonder why that if check is there if the cronjob would restart every time. Thanks for the response.